File: frontend/app/src/main/java/com/example/easypark/easyparkfinal/network/BackendAPI.java
package com.example.easypark.easyparkfinal.network;

import com.example.easypark.easyparkfinal.beans.Categoria;
import com.example.easypark.easyparkfinal.beans.Cliente;
import com.example.easypark.easyparkfinal.beans.LoginDTO;
import com.example.easypark.easyparkfinal.beans.PedidoDTO;
import com.example.easypark.easyparkfinal.beans.PedidoListView;
import com.example.easypark.easyparkfinal.beans.Produto;
import com.example.easypark.easyparkfinal.beans.TokenDTO;
import com.example.easypark.easyparkfinal.beans.Truck;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.Path;


public interface BackendAPI {


    @Headers("Content-Type: application/json")
    @GET("/categoria/listar")
    Call<List<Categoria>> listarCategorias();

    @Headers("Content-Type: application/json")
    @GET("/produto/categoria/{id}")
    Call<List<Produto>> listarProdutosPorCategoria(@Path("id") Long id);

    @Headers("Content-Type: application/json")
    @GET("/produto/truck/{id}")
    Call<List<Produto>> listarProdutosPorTruck(@Path("id") Long id);

    @Headers("Content-Type: application/json")
    @GET("/truck/listar/mesa/{id}")
    Call<List<Truck>> listarTrucksPorMesa(@Path("id") Long id);

    @Headers("Content-Type: application/json")
    @POST("/usuario/cadastrar")
    Call<Boolean> cadastrarUsuario(@Body Cliente cliente);

    @Headers("Content-Type: application/json")
    @POST("/login")
    Call<TokenDTO> logarUsuario(@Body LoginDTO login);

    @Headers("Content-Type: application/json")
    @POST("/pedido/save")
    Call<Boolean> cadastrarPedido(@Body PedidoDTO pedidoDTO);

    @Headers("Content-Type: application/json")
    @GET("/pedido/listar/{id}")
    Call<List<PedidoListView>> listarMeusPedidos(@Path("id") Long id);
}

File: MobileTruck/app/src/main/java/easypark/com/mobiletruck/network/BackendAPI.kt
package easypark.com.mobiletruck.network

import easypark.com.mobiletruck.model.LoginDTO
import easypark.com.mobiletruck.model.PedidoDetalhe
import easypark.com.mobiletruck.model.PedidoOverviewDTO
import easypark.com.mobiletruck.model.Token
import retrofit2.Call
import retrofit2.http.*


interface BackendAPI {

    @POST("/login")
    fun logarUsuario(@Body login: LoginDTO): Call<Token>


    @GET("/pedido/listar/truck/{id}")
    fun lisarPedidos(@Path("id")  id: Long ) : Call<List<PedidoOverviewDTO>>

    @GET("/pedido/detalhe/{id}")
    fun detalharPedido(@Path("id")  id: Long ) : Call<PedidoDetalhe>

    @PUT("/pedido/finalizar/{id}")
    fun finalizarPedido(@Path("id")  id: Long ) : Call<Boolean>

/*
    @Headers("Content-Type: application/json")
    @GET("/produto/categoria/{id}")
    fun listarProdutosPorCategoria(@Path("id") id: Long?): Call<List<Produto>>

    @Headers("Content-Type: application/json")
    @GET("/produto/truck/{id}")
    fun listarProdutosPorTruck(@Path("id") id: Long?): Call<List<Produto>>

    @Headers("Content-Type: application/json")
    @GET("/truck/listar/mesa/{id}")
    fun listarTrucksPorMesa(@Path("id") id: Long?): Call<List<Truck>>

    @Headers("Content-Type: application/json")
    @POST("/usuario/cadastrar")
    fun cadastrarUsuario(@Body cliente: Cliente): Call<Boolean>



    @Headers("Content-Type: application/json")
    @POST("/pedido/save")
    fun cadastrarPedido(@Body pedidoDTO: PedidoDTO): Call<Boolean>

    @Headers("Content-Type: application/json")
    @GET("/pedido/listar/{id}")
    fun listarMeusPedidos(@Path("id") id: Long?): Call<List<PedidoListView>>
    */
}